### PR TITLE
Support jakarta namespace with Inject annotation

### LINKED ISSUE #344: MicroProfile rest client diagnostic using incorrect EE level
Using a project with MP 5 and EE 9 in VS Code (generated with https://start.microprofile.io/)
See the following class, the MP REST client diagnostic appears even though the `@Inject` annotation is already present (but using `jakarta.inject.Inject`)

image

Selecting the quick-fix inserts the `javax.inject.Inject` annotation:
image
image

### PR DESCRIPTION
**Pull request by JessicaJHee**
Peek 2023-01-31 14-03

Fixes #344 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

**Comment by JessicaJHee** [comment]
With a project that uses the `javax` namespace: 
Peek 2023-01-31 14-05

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.test/projects/maven/rest-client-ee9/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.test/projects/maven/rest-client-ee9/pom.xml:1]: I have included a project "open-liberty" that should be almost the same as this project (i.e. it uses the `jakarta` namespace). See if you can get away with just using that one (just to save space).

**Review by datho7561 — CHANGES_REQUESTED** [review]
I think it would be nice to avoid having another test project in the folder, especially given I added one that's fairly similar, but other than that, this PR works very well!

I think it makes sense to split up the PRs for each of the different features that we need to migrate to the jakarta namespace. That way, the PRs are smaller and easier to review, and if you need to shift focus onto other issues, someone else can work on them.

**Comment by angelozerr** [comment]
Your demo are great but I wonder if the import of Inject doesnt exist if the code action insert the proper import.

If no you could do that in a separate pr.

**Comment by JessicaJHee** [comment]
> Your demo are great but I wonder if the import of Inject doesnt exist if the code action insert the proper import.

I believe it should import with the proper namespace based on the project set up. In my demo with a project using the javax namespace, it imports `javax.inject.Inject`.

With a project using `jakarta`, it should work like this: 
Peek 2023-01-31 16-02

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <?pde version="3.8"?>
 <target name="LSP4MP Target Platform">
 	<locations>
-		<!--<location type="Target" uri="https://raw.githubusercontent.com/eclipse/eclipse.jdt.ls/master/org.eclipse.jdt.ls.target/org.eclipse.jdt.ls.tp.target"/>-->
+		<location type="Target" uri="https://raw.githubusercontent.com/eclipse/eclipse.jdt.ls/master/org.eclipse.jdt.ls.target/org.eclipse.jdt.ls.tp.target"/>
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target:5]: Make sure to recomment this line

**Review by datho7561 — APPROVED** [review]
Looks good and works like a charm. Thanks, Jessica!

**Comment by datho7561** [comment]
I'm going to merge this PR but leave #344 open until we handle the rest of the cases where `javax` is used.

**Pull request merged**